syntax = "proto3";

import "google/protobuf/timestamp.proto";

package location;

service LocationService {
  rpc Create(CreateRequest) returns (CreateResponse);
  rpc Retrieve(RetrieveRequest) returns (RetrieveResponse);
}

message Coordinate {
  string latitude = 1;
  string longitude = 2;
}

message Location {
  string id = 1;
  string person_id = 2;
  Coordinate coordinate = 3;
  google.protobuf.Timestamp creation_time = 4;
}

message CreateRequest {
  string person_id = 1;
  Coordinate coordinate = 3;
  google.protobuf.Timestamp creation_time = 4; 
}

message CreateResponse {
  Location location = 1;
}

message RetrieveRequest {}

message RetrieveResponse {
  Location location = 1;
}